### PR TITLE
Custom executor assignment fix

### LINKED ISSUE #1464: Custom Executor Fixes & RTD Update
## Description

The how-to for creating and assigning a custom executor currently does not work as expected because of the recent changes made in order to minimize the usage of pickling/unpickling on covalent server. This actually requires that the custom executor be importable by the `executor_manager` in order for it to be reconstructed and used by covalent.

Furthermore, we want to remove the hardcoded Timing Executor from `covalent/executor/executor_plugins/timing_plugin.py` such that the tutorial and functional test for the how to work by leveraging the updated functionality of custom executors.

## Acceptance Criteria

- [x] Have an interface for the user to specify where their executor plugins can be found by the `executor_manager` (by appending paths to the `COVALENT_EXECUTOR_DIR` env variable)
- [x] Update that how-to guide (https://github.com/AgnostiqHQ/covalent/blob/develop/doc/source/how_to/execution/creating_custom_executors.ipynb) to reflect the above change and how the user can use their custom executors the recommended way
- [x] Remove Timing Executor from `covalent/executor/executor_plugins/timing_plugin.py` 
- [x] Ensure functional test for `creating_custom_executors` passes

### PR DESCRIPTION
**Pull request by kessler-frost**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

Resolves #1464

### CODE REVIEW

**Comment by kessler-frost** [comment]
Ignoring the notebook for functional tests because for some reason manually running the notebook seems to be working but the conversion to script (which is done for the functional test) isn't.

**Review by FyzHsn — CHANGES_REQUESTED** [review]

File: doc/source/how_to/execution/timing_file.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+Node 0 in dispatch dc4b0aa4-13e2-40ad-8ee2-f7e73c769fe0 took 0.00011650000000029692s of CPU time.Node 3 in dispatch dc4b0aa4-13e2-40ad-8ee2-f7e73c769fe0 took 0.00016299999999969117s of CPU time.
```
[the remark this answers is not in the document]
Comment (FyzHsn) [reply doc/source/how_to/execution/timing_file.txt:1]: Or delete it? I don't think this is needed

[the remark this answers is not in the document]
Comment (kessler-frost) [reply doc/source/how_to/execution/timing_file.txt:1]: Agreed, will do 👍

**Review by FyzHsn — APPROVED** [review]

**Review by santoshkumarradha — APPROVED** [review]
good to go, only unblocking comment is to remove final extra input box in jupuyter.

**Pull request merged**